src/components/items.tsx

```diff
@@ -2,51 +2,112 @@ import { useEffect, useState } from "react";
+  useEffect(() => {
+    fetchItems();
   }, []);
   const addItem = () => {
     const $id = uuidv4();
     const newItem = {
       name: input,
       isSelected: false,
     };
-    itemsServices.newItem($id, newItem);
-    setInput("");
+    setThings((prev) => [...prev, { $id, ...newItem }]);
+
+    try {
+      itemsServices.newItem($id, newItem);
+      setInput("");
+    } catch (e) {
+      const newArray = things.filter((i) => i.$id !== $id);
+      setThings(newArray);
+      console.log("Error, Intenta de nuevo", e);
+    }
+  };
+  const updateItem = (t: Partial<TItemDocument>) => {
+    const localUpdate = () => {
+      const newArray = things.filter((i) => i.$id !== t.$id);
+      const index = things.findIndex((i) => i.$id === t.$id);
+      if (index !== -1) newArray.splice(index, 0, t);
+      setThings(newArray);
+    };
+    localUpdate();
+
+    if (t.$id && t.name)
+      try {
+        itemsServices.updateItem(t.$id, {
+          name: t.name,
+          isSelected: t.isSelected || false,
+        });
+      } catch (e) {
+        console.log("Error, Intenta de nuevo", e);
+        localUpdate();
+      }
   };
   const deleteItem = (id: string) => {
-    itemsServices.deleteItem(id);
+    try {
+      itemsServices.deleteItem(id);
+      const newArray = things.filter((i) => i.$id !== id);
+      setThings(newArray);
+    } catch (e) {
+      console.log("Error, no pudo ser eliminado -> Intente nuevamente");
+    }
   };
-  console.log(things);
   return (
     <div className="itemsPage">
-      <div className="inputsField">
+      <Sun className="sunIcon" />
+      <div className="title">
+        <TitleIcon className="elipse1" />
+        <TitleIcon2 className="elipse2" />
+        <h1>Maria's List</h1>
+      </div>
+      <form
+        className="inputsField"
+        onSubmit={(e) => {
+          e.preventDefault();
+          addItem();
+        }}
+      >
         <input
           type="text"
+          maxLength={50}
           value={input}
           onChange={(e) => setInput(e.target.value)}
+          className="heading6"
+          placeholder="Add a new task"
+          required
         />
-        <button onClick={addItem}>Add</button>
-      </div>
+        <button type="submit" className="deleteButtonStyle">
+          <AddItemIcon />
+        </button>
+        <hr className="line"></hr>
+      </form>
+
       <ul>
         {things.map((t) => (
-          <div className="newItem">
-            <input type="checkbox" id="isSelected" />
-            <li>{t.name}</li>
-            <button
-              onClick={() => {
-                deleteItem(t.$id);
-              }}
-            >
-              Delete
-            </button>
-          </div>
+          <Item
+            deleteItem={() => {
+              if (t.$id) deleteItem(t.$id);
+            }}
+            updateItem={(t) => {
+              updateItem(t);
+            }}
+            t={t}
+            key={t.$id}
+          ></Item>
```

Comment:
Puedes cerrarlo solo si el componente no espera `children`
```suggestion
          />
```